### ISSUE TITLE
Calls with nullable warning suppressions cannot be intercepted

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#70061]
**Version Used**: 
```
Microsoft Visual Studio Professional 2022
Version 17.7.4
VisualStudio.17.Release/17.7.4+34031.279
Microsoft .NET Framework
Version 4.8.09037
```

**Steps to Reproduce**:

Compile the following code:
```cs
using System;
using System.Runtime.CompilerServices;

Action? a = null;
a!();
a();

static class InterceptorMethods
{
  [InterceptsLocation("Program.cs", line: 5, character: 1)]
  [InterceptsLocation("Program.cs", line: 6, character: 1)]
  public static void Intercept(this Action a) { }
}

namespace System.Runtime.CompilerServices
{
  [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
  public class InterceptsLocationAttribute : System.Attribute
  {
    public InterceptsLocationAttribute(string path, int line, int character) { }
  }
}
```

**Expected Behavior**:
Either you can intercept both delegate invocations or neither of them.

**Actual Behavior**:
`Error CS9151: Possible method name 'a' cannot be intercepted because it is not being invoked.` is reported for an interceptor that attempts to intercept the call with a nullable warning suppression expression.

It seems that it's completely impossible to intercept the call with a nullable warning suppression as you just get different errors when attempt to change the interception character e.g. to the argument list start offset.
It'd be rather strange for a feature to be incompatible with NRT as you might accidentally break behavior just be suppressing a spurious warning.

Relates to test plan https://github.com/dotnet/roslyn/issues/67421
